src/util/baseutil.js

```diff
@@ -348,3 +348,46 @@ export function isEmpty(value) {
   }
   return true;
 }
+
+/**
+ * Check if IntersectionObserver is supported
+ * @return {boolean} true if window.IntersectionObserver is defined
+ */
+export function isIntersectionObserverSupported() {
+  // Check that 'IntersectionObserver' property is defined on window
+  return typeof window === "object" && window.IntersectionObserver;
+}
+
+/**
+ * Check if native lazy loading is supported
+ * @return {boolean} true if 'loading' property is defined for HTMLImageElement
+ */
+export function isNativeLazyLoadSupported() {
+  return typeof HTMLImageElement === "object" && HTMLImageElement.prototype.loading;
+}
+
+/**
+ * Calls onIntersect() when intersection is detected, or when
+ * no native lazy loading or when IntersectionObserver isn't supported.
+ * @param {Element} el - the element to observe
+ * @param {function} onIntersect - called when the given element is in view
+ */
+export function detectIntersection(el, onIntersect) {
+  if (isNativeLazyLoadSupported() || !isIntersectionObserverSupported()) {
+    // Return if there's no need or possibility to detect intersection
+    onIntersect();
+    return;
+  }
+
+  // Detect intersection with given element using IntersectionObserver
+  const observer = new IntersectionObserver(
+    (entries) => {
+      entries.forEach(entry => {
+        if (entry.isIntersecting) {
+          onIntersect();
+          observer.unobserve(entry.target);
+        }
+      });
+    }, {threshold: [0, 0.01]});
```

Comment:
How was this threshold decided?

Reply:
This is from the vue implementation, in angular you have:
{rootMargin: `0px 0px -50% 0px`, // Margin around the root}
which should it be? @patrick-tolosa

Reply:
In vue our implementation says that as soon as 1% is visible, start loading the image. (That's threshold)

In Angular our implementation says that the image itself is bigger, so we start loading before we reach it.

However, rootMargin is not supported in IE11 and is an experimental feature according to MDN.
I suggest we stick to what you have implemented here, we maybe change it in Angular to match.

Reply:
Thanks @patrick-tolosa 
@strausr can we resolve this?